### PR TITLE
flux controlnet control_guidance_start and control_guidance_end implement

### LINKED ISSUE #9555: [Flux Controlnet] Add control_guidance_start and control_guidance_end
It'd be nice to have `control_guidance_start` and `control_guidance_start` parameters added to flux Controlnet and Controlnet Inpainting pipelines.

I'm currently making experiments with Flux Controlnet Inpainting but the results are poor even with a `controlnet_conditioning_scale` set to 0.6. 

I have to set `controlnet_conditioning_scale` to 0.4 to have non broken results.

Maybe giving more control with the guidance start and end would help reach better results ?

### PR DESCRIPTION
**Pull request by ighoshsubho**
# What does this PR do?

Fixes #9555 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by asomoza** [comment]
thanks a lot, @simbrams can you give this a test and see if it improves your use case?

**Comment by simbrams** [comment]
Thanks a lot @ighoshsubho ! 

> thanks a lot, @simbrams can you give this a test and see if it improves your use case?

I wil be trying it soon, will let you know about the results ASAP.

**Comment by ighoshsubho** [comment]
@simbrams any update on this fix, does it solve what you were looking for? And @yiyixuxu is there any more changes required, if not can this be merged?

**Comment by yiyixuxu** [comment]
@asomoza can you give this a review?

File: src/diffusers/pipelines/flux/pipeline_flux_controlnet.py [hunk 1]
```diff
@@ -572,6 +574,8 @@ def __call__(
         num_inference_steps: int = 28,
         timesteps: List[int] = None,
         guidance_scale: float = 7.0,
+        control_guidance_start: Union[float, List[float]] = 0.0,
+        control_guidance_end: Union[float, List[float]] = 1.0,
```
Comment (asomoza) [inline src/diffusers/pipelines/flux/pipeline_flux_controlnet.py:578]: can you also update the docstring?

File: src/diffusers/pipelines/flux/pipeline_flux_controlnet.py [hunk 2]
```diff
@@ -856,12 +880,19 @@ def __call__(
                 guidance = torch.tensor([guidance_scale], device=device) if use_guidance else None
                 guidance = guidance.expand(latents.shape[0]) if guidance is not None else None
 
+                if isinstance(controlnet_keep[i], list):
+                    current_controlnet_conditioning_scale = [
+                        c * s for c, s in zip(controlnet_conditioning_scale, controlnet_keep[i])
+                    ]
+                else:
+                    current_controlnet_conditioning_scale = controlnet_conditioning_scale * controlnet_keep[i]
```
Comment (asomoza) [inline src/diffusers/pipelines/flux/pipeline_flux_controlnet.py:888]: we should probably stay consistent with the other implementations, is there a reason why you changed this part to be different from the SD3 and SDXL ones?

**Review by asomoza — COMMENTED** [review]
LGTM overall, thanks a lot, left a couple of comments

**Comment by asomoza** [comment]
I took the opportunity to test the jasperai controlnet, here's the example with this applied and a `controlnet_conditioning_scale` of `1.0`:

|default|control_guidance_end=0.5|control_guidance_end=0.5 + control_guidance_start=0.2|
|---|---|---|
||||

File: src/diffusers/pipelines/flux/pipeline_flux_controlnet.py (hunk 1 above)
Comment (ighoshsubho) [reply src/diffusers/pipelines/flux/pipeline_flux_controlnet.py:578]: missed it, sure will do

File: src/diffusers/pipelines/flux/pipeline_flux_controlnet.py (hunk 2 above)
Comment (ighoshsubho) [reply src/diffusers/pipelines/flux/pipeline_flux_controlnet.py:888]: yeah its a mistake, controlnet_conditioning_scale could be a list as well, will make changes

**Comment by simbrams** [comment]
My tests with inpainting are not very conclusive unfortunately, I haven't been able to find parameters that works well with inpainting and "strong" guidance with Depth. If you guys have any tweak ideas that would work to achieve the result bellow, please let me know :)

Inputs:
| Image | Mask | Depth | "Desired" Output |
|--------|--------|--------|--------|
| image | image | image | image Output using SD 1.5 |

Current results: 

`strength`: 0.8
`steps`: 100
`controlnet_conditioning_scale`: 0.8
`controlnet_guidance_start`: 0
`controlnet_guidance_end`: 0.4
image

`strength`: 0.8
`steps`: 100
`controlnet_conditioning_scale`: 0.6 🔻
`controlnet_guidance_start`: 0
`controlnet_guidance_end`: 0.4
image

`strength`: 0.8
`steps`: 100
`controlnet_conditioning_scale`: 0.6
`controlnet_guidance_start`: 0
`controlnet_guidance_end`: 0.6 ↗️
image

`strength`: 0.8
`steps`: 100
`controlnet_conditioning_scale`: 0.4 🔻
`controlnet_guidance_start`: 0
`controlnet_guidance_end`: 0.8 ↗️
image

`strength`: 0.8
`steps`: 100
`controlnet_conditioning_scale`: 0.5 ↗️
`controlnet_guidance_start`: 0
`controlnet_guidance_end`: 0.8
image

**Comment by burgalon** [comment]
any chance to get this merged?

**Comment by asomoza** [comment]
@simbrams this just confirms what I was thinking and saw, Flux right now is not a good solution for inpainting, maybe it will get better with time but the model since it's distilled and very forceful aesthetic might be a problem with realistic photo inpainting.

You have a very little margin to work with it and it takes a long time to be able to experiment with it and even more to train a decent controlnet. 

Maybe you can try with the recent "undistilled" models? I think there's five of them right now, they will take more time for inference since they have CFG but probably will yield better results.

**Review by asomoza — APPROVED** [review]

**Comment by simbrams** [comment]
> @simbrams this just confirms what I was thinking and saw, Flux right now is not a good solution for inpainting, maybe it will get better with time but the model since it's distilled and very forceful aesthetic might be a problem with realistic photo inpainting.
> 
> You have a very little margin to work with it and it takes a long time to be able to experiment with it and even more to train a decent controlnet.
> 
> Maybe you can try with the recent "undistilled" models? I think there's five of them right now, they will take more time for inference since they have CFG but probably will yield better results.

Indeed you were right ! I will try with undistilled models and see how it goes :) thanks for the tip!

**Comment by asomoza** [comment]
oh I forgot about it, they need the CFG so they probably only work with the [community pipeline](https://github.com/huggingface/diffusers/blob/main/examples/community/pipeline_flux_with_cfg.py)  and currently we don't have a controlnet pipeline for Flux with CFG. 

You can open an issue about and we can discuss it there, I'll merge this PR now.

**Comment by asomoza** [comment]
Thanks a lot @ighoshsubho, the failing test is unrelated.
